### PR TITLE
Make MMM Plots tests run faster

### PR DESCRIPTION
**Pull request by juanitorduz**
Closes: https://github.com/pymc-labs/pymc-marketing/issues/361

----
:books: Documentation preview :books:: https://pymc-marketing--363.org.readthedocs.build/en/363/

### CODE REVIEW

File: tests/mmm/test_plotting.py [hunk 1]
```diff
@@ -62,19 +62,21 @@ class ToyMMM(BaseDelayedSaturatedMMM, MaxAbsScaleTarget):
...
-                adstock_max_lag=4,
+                adstock_max_lag=2,
                 control_columns=["control_1", "control_2"],
                 channel_columns=["channel_1", "channel_2"],
             )
         # fit the model
         mmm.fit(
             X=toy_X,
             y=toy_y,
+            chains=2,
```
[the remark this answers is not in the document]
Comment (juanitorduz) [reply tests/mmm/test_plotting.py:78]: Actually, I think the bottleneck comes from https://github.com/pymc-labs/pymc-marketing/blob/main/pymc_marketing/model_builder.py#L308 as we are not use the `samples` argument in the `plot_prior` which is one of the slowest tests. @michaelraczycki can we add `**kwargs` to be also past to the `sample_prior_predictive`?

[the remark this answers is not in the document]
Comment (ricardoV94) [reply tests/mmm/test_plotting.py:79]: Is it really wise to call all 3 sampling method when "fitting" the model?

[the remark this answers is not in the document]
Comment (juanitorduz) [reply tests/mmm/test_plotting.py:79]: I guess this discussion might be beyond this PR as it is more about the `fit` scope in the `ModelBuilder`.  I suggest we merge these small changes which make the ci time from 33min -> 27 min and then iterate on the prior_predictive_sample arguments. WDYT?

[the remark this answers is not in the document]
Comment (ricardoV94) [reply tests/mmm/test_plotting.py:79]: Sure that is out of scope.

I still don't think we should be doing mcmc sampling at all just for a plotting test. Definitely don't need prior/posterior and posterior predictive draws?

[the remark this answers is not in the document]
Comment (juanitorduz) [reply tests/mmm/test_plotting.py:79]: > I still don't think we should be doing mcmc sampling at all just for a plotting test. 
ok! so I then suggest we close this PR and I will open another one where I try to do it with dummy idata as suggested.

**Comment by juanitorduz** [comment]
I will try to avoid sampling at all for the plotting tests in a future PR.

File: tests/mmm/test_plotting.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (michaelraczycki) [reply tests/mmm/test_plotting.py:79]: I'll make the additional sampling  (prior and posterior predictive) optional, controlled by boolean parameters. That way we can make it more compact by people that actually want to get all samples straight away, and not force anyone to burn computing time if they don't want to. Fine with you? @juanitorduz off, I'll include it in the same PR, I'm also adjusting the docstrings in model_builder, as they need some more attention

**Pull request closed without merging**